zebra/interface.c

```diff
@@ -1977,6 +1977,8 @@ DEFUN (shutdown_if,
 	struct zebra_if *if_data;
 
 	if (ifp->ifindex != IFINDEX_INTERNAL) {
+		/* send RA lifetime of 0 before stopping. rfc4861/6.2.5 */
```

Comment:
Code changes looks good. I will approve it. 

Quick question: On interface shutdown,  the new code is setting the "nd_ra_router_lifetime" to zero. Now this attribute is set to configured "pkt_RouterLifetime" only when rt_adv_timer is expired and rtadv_send_packet is called with RA_ENABLE. Is this expected or do we need to set this to the configured value on interface UP ?

Reply:
We are only setting the value in the send_packet function in the case we are ceasing to send RAs and it only effects the current packet sent.  We aren't changing the stored value of what nd_ra_router_lifetime should be so when the interface comes back up, it retains the defined (or default) value.  I've tested this extensively and it works properly on if up (or other state changes that cause RAs to flow).

Reply:
Got it. Thanks.